### ISSUE TITLE
Queue on Linux never gets longer than 1

### ISSUE THREAD

**Issue by kradalby** [FabrizioBrancati/Queuer#19]
Hi

I have been experimenting with this library for a tool I am porting from macOS to Linux and I seem to have a strange problem that I am unsure if is in this library or underlying in dispatch.

I am trying to add work to a queue on Linux to parallelise it, but it looks like the queue is never longer than one item and therefore not doing anything in parallel.

A minimum example:

```swift
import Queuer
import Foundation
import Dispatch


let queue = Queuer(
        name: "write", maxConcurrentOperationCount: Int.max, qualityOfService: .background)

for i in 1 ..< 10 {
        let op = ConcurrentOperation { _ in
                Thread.sleep(forTimeInterval: 0.2)
                print(i)
        }
        // queue.addOperation(op)
        op.addToQueue(queue)
        print("Queue length: ", queue.operationCount)
}

print("Queue length: ", queue.operationCount)
queue.waitUntilAllOperationsAreFinished()
print("Queue length: ", queue.operationCount)
```

Resulting in (where everything _always_ print in order, one at the time):
```
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
1
2
3
4
5
6
7
8
9
Queue length:  0
```

I might be doing something silly, but I am struggling to see something obvious.

EDIT:

Essential information I forgot:

I have tested this with Queuer 2.1.1 and Swift 5.2.4 on Ubuntu 20.04 and 20.10 Beta. (and macOS, but that works as intended)

**Comment by FabrizioBrancati** [comment]
Queuer has some unit tests, but this case is not [covered here](https://github.com/FabrizioBrancati/Queuer/blob/master/Tests/QueuerTests/QueuerTests.swift#L32).
We should add a test for that case, maybe we will understand the cause of this bug.

Maybe the bug is on Dispatch for Ubuntu 20.xx or maybe on Swift 5.2.x... On Travis, I used Swift 5.1 for Linux because of a `segfault 11` error on Swift 5.2.x.

**Comment by kradalby** [comment]
I can give 5.1 a go after work to try to narrow it down. Did you use Ubuntu 20.xx or 18.04/16.04?

**Comment by FabrizioBrancati** [comment]
Currently, Travis doesn't have Ubuntu 20.04, they only have 18.04 and 16.04

**Comment by kradalby** [comment]
I just managed to confirm it is present on Ubuntu 18.04 with Swift 5.1.5:

```
root@swift3:~/git/DispatchTest# swift -version
Swift version 5.1.5 (swift-5.1.5-RELEASE)
Target: x86_64-unknown-linux-gnu
root@swift3:~/git/DispatchTest# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.4 LTS
Release:        18.04
Codename:       bionic
root@swift3:~/git/DispatchTest# swift run
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
Queue length:  1
1
2
3
4
5
6
7
8
9
Queue length:  0
```

Also tested 20.04 with 5.1.5

**Comment by FabrizioBrancati** [comment]
Do you mind opening a PR with a new unit test to test this case? We will see what happens on Travis with Ubuntu 16.04 and 18.04

**Comment by kradalby** [comment]
I'll give it a go, sorry I didnt update, I played around a bit during the weekend and it seems that this is a problem with OperationQueues in Linux. I switched to DispatchQueues which works fine on Linux, but has less high level controls.

**Comment by FabrizioBrancati** [comment]
This is now fixed on Queuer 3.0.0 🎉 Thanks!
